ruma-client-api/src/r0/capabilities/iter.rs

```diff
@@ -0,0 +1,72 @@
+//! Iterator implementation for `Capabilities`
+
+use std::{borrow::Cow, collections::btree_map};
+
+use serde_json::Value as JsonValue;
+
+use super::Capabilities;
+
+/// Reference to a capability.
+#[derive(Debug)]
+pub struct CapabilityRef<'a> {
+    name: &'a str,
+    value: Option<&'a JsonValue>,
+    caps: &'a Capabilities,
+}
+
+impl<'a> CapabilityRef<'a> {
+    /// Get name of the capability.
+    pub fn name(&self) -> &'a str {
+        self.name
+    }
+
+    /// Get value of the capability.
+    pub fn value(&self) -> Cow<'a, JsonValue> {
+        match self.value {
+            // unknown capability from btreemap iterator
+            Some(val) => Cow::Borrowed(val),
+            // O(1) lookup of known capability
+            None => self.caps.get(self.name).unwrap(),
+        }
+    }
+}
+
+/// An iterator over capabilities.
+#[derive(Debug)]
+pub struct CapabilitiesIter<'a> {
+    /// Reference to Capabilities
+    pub caps: &'a Capabilities,
+    /// Current position of the iterator
+    pub pos: usize,
+    /// Iterator for custom capabilities
+    pub custom_caps_iterator: btree_map::Iter<'a, String, JsonValue>,
```

Comment:
My suggestion was to make these fields private again. They shouldn't need to be public for anything, right?

Reply:
Oh  yes